airflow/api/common/experimental/trigger_dag.py

```diff
@@ -86,7 +86,10 @@ def trigger_dag(
         execution_date=None,
         replace_microseconds=True,
 ):
-    dagbag = DagBag()
+    dag_model = DagModel.get_current(dag_id)
+    if dag_model is None:
+        raise DagNotFound("Dag id {} not found in DagModel".format(dag_id))
+    dagbag = DagBag(dag_folder=dag_model.fileloc)
```

Comment:
That looks simple. I take it since you opened a PR this helped?

Reply:
I checked with triggering a DAG where we had 500 total number of dags.
Latency reduced from 10 sec to 1.5 sec.